File: src/main/java/entity/Category.java
package entity;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;

import java.util.*;

import com.fasterxml.jackson.annotation.JsonIgnore;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Category {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer categoryId;
	private String categoryCode;
	private String categoryName;
	private String categoryImagePath;
	private Boolean flag;
	
	@OneToMany(mappedBy = "category",cascade = CascadeType.ALL)
	@JsonIgnore
	private List<Subcategory> subcategories;
	
}

File: DotNet_Backend/SafarnamaApplication/SafarnamaApplication/Models/Category.cs
using System;
using System.Collections.Generic;

namespace SafarnamaApplication.Models;

public partial class Category
{
    public int CategoryId { get; set; }

    public string? CategoryCode { get; set; }

    public string? CategoryName { get; set; }

    public string? CategoryImagePath { get; set; }

    public bool? Flag { get; set; }

    public virtual ICollection<Package> Packages { get; set; } = new List<Package>();

    public virtual ICollection<Subcategory> Subcategories { get; set; } = new List<Subcategory>();
}
